app/code/RLTSquare/Unit3/Block/CustomerDetails.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RLTSquare\Unit3\Block;
+
+use Magento\Framework\Encryption\EncryptorInterface;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+use RLTSquare\Unit3\Helper\Data as Helper;
+
+/**
+ * block class for fetching customer information
+ */
+class CustomerDetails extends Template
+{
+    /**
+     * @var Helper
+     */
+    private Helper $helper;
+    /**
+     * @var EncryptorInterface
+     */
+    private EncryptorInterface $encryptor;
+
+    /**
+     * constructor
+     *
+     * @param Context $context
+     * @param Helper $helper
+     * @param EncryptorInterface $encryptor
+     */
+    public function __construct(Context $context, Helper $helper, EncryptorInterface $encryptor)
+    {
+        $this->helper = $helper;
+        $this->encryptor = $encryptor;
+        parent::__construct($context);
+    }
+
+    /**
+     * method to get username from helper
+     *
+     * @return string
+     */
+    public function userName(): string
```

Comment:
The "userName" function name does not indicate whether function reads or sets the value ... the "getUsername" would be more indicative.